lib/generators/newrelic/setup_generator.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Newrelic
+  class SetupGenerator < Rails::Generators::Base
+    NEWRELIC_GEM_NAME = 'newrelic_rpm'
+    NEWRELIC_GEM_USED_REGEXP = /gem.*\b#{NEWRELIC_GEM_NAME}\b.*/
+
+    desc 'This generator adds the newrelic gem and creates all the needed config files.'
+    argument :app_name, type: :string, required: true
+
+    def add_new_relic_gem
+      gem(NEWRELIC_GEM_NAME) unless newrelic_installed?
```

Comment:
does this `gem` command pin the gem to a certain version?

Reply:
Not right now, but we can provide that information to this method.
```ruby
gem(NEWRELIC_GEM_NAME), version: '~> 1.1'
```

I'm not sure if that's something we really want to do. Setting a version here means that we need to track every gem version change and manually update this file. In this scenario, we don't have tools like dependabot that can help us with the upgrade so it's kinda difficult to maintain.

I was thinking this could be a manual step. If we enable/create a cop that forces all the gems to have a version specified in the Gemfile, devs will need to do it anyway.